### PR TITLE
Reimplement buttons

### LINKED ISSUE #12: ffe-buttons-react: TertiaryButton only supports rightIcon if there is also a leftIcon
`TertiaryButton` doesn't support an icon on the right side of the label unless there is also an icon on the left side. This is an unintended effect of the `simpleContent` prop in `Button` being set to `true` (which renders a simpler markup without support for icons.) unless there is a `leftIcon`.

Quickfix: set `simpleContent` to `false` if there is either a `leftIcon` or a `rightIcon` ([here](https://github.com/SpareBank1/designsystem/blob/e7b68cd44af010bd9e7debc621137f3a6fe84874/packages/ffe-buttons-react/src/TertiaryButton.js#L19))

Perhaps better: get rid of the whole `simpleContent` concept and restructure the components internally to support the different use cases in a simpler way

### PR DESCRIPTION
**Pull request by selbekk**
This pull request re-implements both `ffe-buttons` and `ffe-buttons-react` in a long-awaited major breaking change.

Please give me feedback on the way I've solved things, as I want to try to find the best possible API before merging it.

Fixes #12 and #4

### CODE REVIEW

File: packages/ffe-buttons-react/src/ExpandButton.js
Comment (wkillerud) [inline packages/ffe-buttons-react/src/ExpandButton.js]: If I provide a `leftIcon` I would expect it to apply only to the label in its unexpanded state. Now it will override the `KryssIkon` when `isExpanded` is `true`. `KryssIcon` should perhaps instead be rendered instead of `children`?

That being said, `ExpandButton` should probably not build on `BaseButton` here in React IMO. We don't have an `isLoading` state, and so rendering the spinner and its wrapper seems a bit pointless.

Comment (wkillerud) [reply packages/ffe-buttons-react/src/ExpandButton.js]: `InlineBaseButton` would be a better parent I think :) Only, this isn't necessarily inline. Hmm..

File: packages/ffe-buttons/less/base-button.less
Comment (wkillerud) [inline packages/ffe-buttons/less/base-button.less:35]: Unintended, or should the default and focus state look the same?

**Review by wkillerud — CHANGES_REQUESTED** [review]
I'd love the commit messages even more if they had a "before" example as well - how does the markup look today and what are the key changes? (no modifiers inside the button, no label wrapper, but a wrapper for the spinner instead).

I'd also love it if the commit message for the breaking change in React-land mentions the most important breaking changes instead of pointing to the .md immediately.

File: packages/ffe-buttons/less/base-button.less
Comment (selbekk) [reply packages/ffe-buttons/less/base-button.less:35]: According to the [original implementation](https://github.com/SpareBank1/designsystem/blob/master/packages/ffe-buttons/less/action-button.less#L16-L21), this is how it should look.

File: packages/ffe-buttons-react/src/ExpandButton.js
Comment (selbekk) [reply packages/ffe-buttons-react/src/ExpandButton.js]: Some good points! 
The reason why this abstraction was used is because the `ffe-button--expand` extends the look and feel of the `ffe-button--secondary`. It shares the same common less-class as well (`ffe-button`), and it follows a lot of the same usage patterns.

As you might have noticed, I added more full-featured prop types to each of the components than they had previously. This is to specify the supported API better than what the previous implementation did. I considered destructuring waay unsupported properties so that misuse would be even harder, but decided against it. I believe undocumented usage should be discouraged, not defensively coded against.

That being said, I'll try to implement a custom component that tries to solve this a bit better.

**Comment by selbekk** [comment]
Okay, so I did a few changes:

- Improved the commit logs for easier migration
- Simplified the spinner implementation, now it only needs a single DOM element
- Renamed prop `buttonRef` to `innerRef` (which is a bit easier to standardize across the codebase)
- Re-added `ariaLoadingMessage` as a documented prop, but with a sensible default.
- A few documentation improvements

**Review by wkillerud — APPROVED** [review]

**Review by henrikhermansen — APPROVED** [review]
Ser veldig bra ut :1st_place_medal:

**Comment by selbekk** [comment]
I would really love a review from one of the core maintainers (@kwltrs @antidecaf) on this before it's merged. The changes in the LESS API is big and breaking, so I would love to have most people on board 👯‍♂️

**Review by antidecaf — APPROVED** [review]
LGTM!

**Comment by selbekk** [comment]
@kwltrs or @antidecaf  do you mind merging this whenever you're ready to release it? Since it's a complete rewrite, I think it's beneficial if we try to release it as soon as possible after a merge to master.

**Review by kwltrs — APPROVED** [review]

**Pull request merged**
